src/lightning/fabric/utilities/cloud_io.py

```diff
@@ -24,6 +24,21 @@
 
 from lightning.fabric.utilities.types import _MAP_LOCATION_TYPE, _PATH
 
+try:
+    import adlfs
+except ImportError:
+    adlfs = None
+
+try:
+    import gcsfs
+except ImportError:
+    gcsfs = None
+
+try:
+    import s3fs
+except ImportError:
+    s3fs = None
```

Comment:
I suggest we move these to a local import inside the function. Reason: These are not dependencies, and someone might import lightning with them installed yet doesn't use the functionality below. So we should attempt to minimize the import time (#12786) via importing them lazily.

Reply:
Agreed, that's a great feedback, I moved the imports inside the `_is_object_storage` function